### PR TITLE
Exception Logger

### PR DESCRIPTION
**Pull request by JulianForeman**
# Test Links:
[Percentile Calculator](https://wps-pr-1808.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-1808.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-1808.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-1808.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-1808.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=1108&f=c5&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN#state=2ec784ca-c46a-49d0-b2b3-1cf32a9015a2&session_state=7d9447c8-db66-4661-b4cb-03d2ac0d1d8f&code=32292df4-2bdf-4f90-a4a8-c8dbcda682a9.7d9447c8-db66-4661-b4cb-03d2ac0d1d8f.2b63f390-f3dc-43ae-89f2-016453863476)
[Fire Behaviour Advisory](https://wps-pr-1808.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-1808.apps.silver.devops.gov.bc.ca/hfi-calculator)
[FWI Calculator](https://wps-pr-1808.apps.silver.devops.gov.bc.ca/fwi-calculator)

### CODE REVIEW

File: api/app/main.py [hunk 1]
```diff
@@ -80,6 +81,16 @@
 
 ORIGINS = config.get('ORIGINS')
 
+
+async def catch_exception_middleware(request: Request, call_next):
+    try:
+        return await call_next(request)
+    except Exception as exc:
+        logger.debug(exc, exc_info=True)
```
Comment (Sybrand) [inline api/app/main.py:89]: Log it as error please - not debug.
```suggestion
        logger.error(exc, exc_info=True)
```

**Review by Sybrand — CHANGES_REQUESTED** [review]
That's delightfully easy! Awesome. Just make it logger.error pls.

**Review by Sybrand — APPROVED** [review]

**Review by Sybrand — CHANGES_REQUESTED** [review]
Failing on lint (need comments for the new function) - otherwise looking good.

File: api/app/main.py (hunk 1 above)
Comment (conbrad) [reply api/app/main.py]: @Sybrand shouldn't we be posting this to the RC channel?

Comment (conbrad) [reply api/app/main.py]: Any unhandled exception that bubbles up to this level seems quite severe

**Review by Sybrand — APPROVED** [review]

File: api/app/main.py (hunk 1 above)
Comment (Sybrand) [reply api/app/main.py]: At this point going to rocket chat server no purpose, given the sheer amount of garbage logged there - it's essentially useless.
That's a separate concern though. I agree - logging to rocket chat would be a good idea.

Comment (conbrad) [reply api/app/main.py]: I agree that RC is noisy but I think it's better than log output that disappears with ephemeral pods. I still use RC messages to troubleshoot and be alerted to issues.

Comment (conbrad) [reply api/app/main.py]: Also, won't a raised error here cause the pod to shutdown anyways, making this log useless?

**Pull request merged**